.orchestra/config/components/nanobind.yml

```diff
@@ -0,0 +1,34 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:overlay", "overlay")
+#@ load("/lib/create-component.lib.yml", "single_build_component")
+#@ load("/lib/shell.lib.yml", "expand_args")
+#@ load("/lib/cmake.lib.yml", "cmake", "cmdline_cmake_base_configuration")
+#@ load("/lib/optimization-flavors.lib.yml", "typical_cmake_flavors")
+
+#@yaml/text-templated-strings
+---
+#@ def args():
+license: LICENSE
+repository: nanobind
+configure: |
+  git -C "$SOURCE_DIR" submodule update --init --recursive
+  mkdir -p "$BUILD_DIR"
+  cd "$BUILD_DIR"
+  
+  (@= cmake @) \
+    (@= expand_args(cmdline_cmake_base_configuration(cmake_build_type="Release", extra_compiler_flags="-ggdb3 -O2 -fno-omit-frame-pointer")) @) \
+    "$SOURCE_DIR/standlone" \
+    -DBUILD_SHARED_LIBS=ON
```

Comment:
Use `typical_cmake_builds`, it should be flexible enough to do everything you need to do.

Reply:
`typical_cmake_builds` introduces `@release` and `@debug` which are not useful. I've gone a bit overboard on this because `gdb` wasn't showing me location info for `libnanobind.so`, but I'm also getting them for llvm, so it might be my `gdb` being misconfigured.

Reply:
> typical_cmake_builds introduces @release and @debug which are not useful

I know, still, do what others are doing. When we refactor, these things done by hand are free extra suffering.

> I've gone a bit overboard on this because gdb wasn't showing me location info for libnanobind.so, but I'm also getting them for llvm, so it might be my gdb being misconfigured.

Not sure what you mean here.
How is this related to `typical_cmake_builds`?

Reply:
Replaced things with `typical_cmake_builds`